SmartTutor/Controllers/ChallengeController.cs

```diff
@@ -0,0 +1,21 @@
+﻿using SmartTutor.ActiveEducationModel;
+using SmartTutor.Repository.ChallengeRepository;
+using SmartTutor.Service;
+
+namespace SmartTutor.Controllers
+{
+    public class ChallengeController
+    {
+        public ChallengeService ChallengeService;
+
+        public ChallengeController()
+        {
+            ChallengeService = new ChallengeService(new ChallengeRepository());
+        }
+
+        public void StartChallenge(SmellType issue, int indexOfProject, Player player)
```

Comment:
This controller looks like it will interface with the IDE plugin.

Since we will look at challenges as learning objects that are part of the "Procedural knowledge" (PK) knowledge node (KN), the start challenge would load the KN with the given ID (so the plugin would also need a function to get Lectures and then get PK KNs for the selected lecture - this can be a single function that sends KN DTOs). **Note that challenges are related to Lectures, not smells**.